crates/virtio-queue/src/state.rs

```diff
@@ -149,6 +150,77 @@ impl QueueState {
             .map(Wrapping)
             .map_err(Error::GuestMemory)
     }
+
+    /// Used in the test_invalid_avail_idx from queue.rs and test_queue_guard_object from
+    /// queue_guard.rs in order to de-duplicate the tests
+    pub fn test_state<M: GuestMemory>(&mut self, mem: &M, vq: &MockSplitQueue<GuestMemoryMmap>) {
```

Comment:
Since this is not a test, but a test helper, it should be named starting with `check_` or something similar.

We should name this more appropriate to what it's doing because in the future we will add more common checks here that can be re-used in queue and queue guard. A good name would be `check_invalid_avail_idx`, and for the comment we can write that this is a test case that checks that an invalid `avail_idx` is not triggering a panic in the queue implementation.

Reply:
Done